bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -4023,6 +4023,7 @@ void ReplicaImp::executeRequestsAndSendResponses(PrePrepareMsg *ppMsg,
 }
 
 void ReplicaImp::tryToRemovePendingRequestsForSeqNum(SeqNum seqNum) {
+  if (lastExecutedSeqNum >= seqNum) return;
   SCOPED_MDC_SEQ_NUM(std::to_string(seqNum));
```

Comment:
@yontyon , could you explain in what situation this might happen? Is it in some specific situation, as far as I know the common case is to first commit and then execute?

Reply:
A replica may receive a commit message even after it executed the same request (say, if the message has been delayed).
In this case, if we don't check it elsewhere we may clean again the client pending requests of this sequence number requests (maybe we do prevent this case even now, but I think it is better to make sure)

Reply:
@HristoStaykov do you agree with this comment?

Reply:
Yes I agree, it is good to have this check to make sure we have correct behaviour over all code paths.